robowflex_ompl/include/robowflex_ompl/ompl_interface.h

```diff
@@ -93,6 +95,14 @@ namespace robowflex
              */
             ompl_interface::OMPLInterface &getInterface() const;
 
+            /** \brief Set a callback, to be called right before a planning session for last-minute
+             * configuration or external bookkeeping.
+             *
+             * refreshContext() will already have been called, so the SimpleSetup obtained by
+             * getLastSimpleSetup() will be the one used for planning.
+             */
+            void setPreplanCallback(const std::function<void()> &preplanCallback);
```

Comment:
Two things:
1. It would probably be useful to also pass in the requested scene and request (as const) to the callback function.
2. You should create an alias with `using` to define the callback function type.

Reply:
For simplicity's sake, you might also want to pass in the simple setup or planning context as well as non-const arguments to the callback.

Reply:
The scene and request makes sense. The SimpleSetup would make sense too, since the method would be clearer in terms of that actually being the SImpleSetup that's being used. Maybe the planning context too, which contains the SimpleSetup?

As I said in a comment down below, in my case I just capture reference to the planner in the lambda that I'm supplying to the callback, though that wouldn't quite as well if you're just passing in a function pointer.

Reply:
I've added the args and factored the type out. I figured I'd pass in the whole planning context since it's at least as useful as the SimpleSetup, and suffers from the same problem of not being easily accessible (without taking over the control flow in a slightly awkward way) from outside the planner despite that access being quite useful.